src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -9,13 +10,28 @@ function div_hUv(i, j, k, grid, advection, solution)
 end
 
 @inline momentum_flux_huu(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uu(i, j, k, grid, solution.uh, solution.uh) / h[i, j, k]
+    @inbounds momentum_flux_uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
 
 @inline momentum_flux_huv(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uv(i, j, k, grid, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
+    @inbounds momentum_flux_uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
 @inline momentum_flux_hvu(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_vu(i, j, k, grid, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
+    @inbounds momentum_flux_vu(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
```

Comment:
Similar problem as above regarding high-order interpolation of `uh`, `vh`, combined with second-order interpolation of `h`.

Reply:
Echoing an above comment: could make sense to save fixing `h` interpolation for a future PR.

Reply:
Agreed